dojo/settings/settings.dist.py

```diff
@@ -844,6 +846,7 @@ def generate_url(scheme, double_slashes, user, password, host, port, path, param
     'Anchore Engine Scan': DEDUPE_ALGO_HASH_CODE,
     'Checkmarx Scan detailed': DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL,
     'Checkmarx Scan': DEDUPE_ALGO_HASH_CODE,
+    'Checkmarx OSA': DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL_OR_HASH_CODE,
```

Comment:
As you always set uniq id, you should switch to DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL
```suggestion
    'Checkmarx OSA': DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL,
```

Reply:
although the unique_id will always be there, the hash_code can be used for cross-parser deduplication. this is still experimental and will probably require a bit of standardization, still this configuration is the first step :)

Reply:
Regarding this specific topic, as I know more well parsers, I really think that relaying on the hash code for cross-parser de-duplication will be really hard. Most of the parsers use too much specific data for the hash code like titles that are very different between parsers.
The only reliable columns that could help us to do that will be **CVE**.

Reply:
some parsers have cve, others have cwe. you also need to add the component name, or file path to identify the vulnerability because CVE only will match many different findings. For checkmarx OSA i have choose the component name and version, which should be more or less standard and could be the same as other parsers (I haven't checked).  All this information that makes it possible to identify a vulnerability should  go into the hash_code so that cross-parser deduplication can maybe work one day. I agree that many parsers have a poorly configured hash_code but that can always be changed later. 

 I think CVE only cannot work so I see only two solutions: 
1) use the hash_code
2) add another hash_code or something similar. specifically for cross-parser dedupe. that might make sense, we could discuss this but we'll need to change the deduplication to use that

Reply:
It's very interesting to understand your point of view! Thanks for sharing!
2 things I saw when I deep dive into parsers code:
1. component_name is very new and most of parser imp were wrong. (sometime mixing vendor/name/version in this attribute). I fixed few but for now every parser use it in different ways. Less reliable than CVE :/
2. parsers implement things in very different ways. for ex title are implemented differently even with the same attributes! So different implementation of fields will lead to completely different and incomparable hash codes. we will need a document that define what the "perfect" hash code should be, and what the attributes should contains and fix every parsers to follow this spec to have something that follow this spec. near impossible goal IMO.

Regarding component name and version, the work of fixing them is not done.